packages/components/src/internal/components/domainproperties/samples/UniqueIdBanner.spec.tsx

```diff
@@ -0,0 +1,219 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { ADD_NEW_UNIQUE_ID_MSG, NEW_TYPE_NO_BARCODE_FIELDS_MSG, UniqueIdBanner } from './UniqueIdBanner';
+import { SampleTypeModel } from './models';
+import { Alert } from '../../base/Alert';
+import { Button } from 'react-bootstrap';
+import { INT_RANGE_URI, STORAGE_UNIQUE_ID_CONCEPT_URI, STRING_RANGE_URI } from '../constants';
+import { DomainDesign, DomainDetails, DomainField } from '../models';
+import { fromJS } from 'immutable';
+
+const newSampleTypeModel = SampleTypeModel.create();
+
+const intField = {
+    name: 'key',
+    rangeURI: INT_RANGE_URI,
+    propertyId: 1,
+    propertyURI: 'test',
+};
+
+const stringField = {
+    name: 'string',
+    rangeURI: STRING_RANGE_URI,
+    propertyId: 2,
+    propertyURI: 'test',
+};
+
+
+const uniqueIdField = {
+    name: 'barcode',
+    rangeURI: STRING_RANGE_URI,
+    conceptURI: STORAGE_UNIQUE_ID_CONCEPT_URI,
+    propertyId: 3,
+    propertyURI: 'test'
+};
+
+const uniqueIdField2 ={
+    name: 'Other Barcode',
+    rangeURI: STRING_RANGE_URI,
+    conceptURI: STORAGE_UNIQUE_ID_CONCEPT_URI,
+    propertyId: 3,
+    propertyURI: 'test'
+};
+
+describe("<UniqueIdBanner/>", () => {
+    test("new, in properties panel, no uniqueID fields", () => {
+        const wrapper = mount(
+            <UniqueIdBanner
+                model={newSampleTypeModel}
+                isFieldsPanel={false}
+                onAddField={jest.fn}
+            />
+        );
+        expect(wrapper.text()).toBe(NEW_TYPE_NO_BARCODE_FIELDS_MSG);
```

Comment:
any reason to check to make sure no alert or buttons show up? i.e. the opposite of the check in the test case below.